{
  "comments": [
    {
      "key": {
        "uuid": "6c621a39_157f2a2e",
        "filename": "components/ml-prediction-ms/config/dmaap/MsgRtrApi.properties",
        "patchSetId": 38
      },
      "lineNbr": 2,
      "author": {
        "id": 4405
      },
      "writtenOn": "2023-03-28T22:12:19Z",
      "side": 1,
      "message": "Is this file still needed?\nCan we remove all refs to dmaap?",
      "range": {
        "startLine": 2,
        "startChar": 12,
        "endLine": 2,
        "endChar": 17
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "27e41a64_72dd8f42",
        "filename": "components/ml-prediction-ms/config/dmaap/MsgRtrApi.properties",
        "patchSetId": 38
      },
      "lineNbr": 2,
      "author": {
        "id": 8012
      },
      "writtenOn": "2023-03-29T12:45:03Z",
      "side": 1,
      "message": "File removed",
      "parentUuid": "6c621a39_157f2a2e",
      "range": {
        "startLine": 2,
        "startChar": 12,
        "endLine": 2,
        "endChar": 17
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "16905185_067941f0",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 168,
      "author": {
        "id": 4405
      },
      "writtenOn": "2023-03-28T22:12:19Z",
      "side": 1,
      "message": "Ideally we should remove any references to dmaap.",
      "range": {
        "startLine": 168,
        "startChar": 8,
        "endLine": 168,
        "endChar": 13
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "8082b4a2_be38cd1e",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 168,
      "author": {
        "id": 8012
      },
      "writtenOn": "2023-03-29T12:44:31Z",
      "side": 1,
      "message": "Resolved",
      "parentUuid": "16905185_067941f0",
      "range": {
        "startLine": 168,
        "startChar": 8,
        "endLine": 168,
        "endChar": 13
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "edced68b_abb16539",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 565,
      "author": {
        "id": 4405
      },
      "writtenOn": "2023-03-28T22:12:19Z",
      "side": 1,
      "message": "Not sure if this is planned to be run as a standalone mS in ONAP but if so the Producer and Consumer conf will need to be extended to support additional security/authentication params.",
      "range": {
        "startLine": 565,
        "startChar": 12,
        "endLine": 565,
        "endChar": 16
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": true
    },
    {
      "key": {
        "uuid": "e5aa0eea_ceec9e75",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 565,
      "author": {
        "id": 8012
      },
      "writtenOn": "2023-03-29T12:44:31Z",
      "side": 1,
      "message": "As per my understanding, It is not planned to run in standalone mode. currently it is interacting with external Kafka consumer and producer related to PERFORMANCE_MEASUREMENTS and Policy Topics. When we try to add any sasl_mechanism we can see error messages at Kafka service. So currently i believe security/ authentication are not implemented at external ends.",
      "parentUuid": "edced68b_abb16539",
      "range": {
        "startLine": 565,
        "startChar": 12,
        "endLine": 565,
        "endChar": 16
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "92d323a0_44cf7e33",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 565,
      "author": {
        "id": 4405
      },
      "writtenOn": "2023-03-29T12:51:59Z",
      "side": 1,
      "message": "In ONAP we are using Strimzi kafka with the following security requirements for all clients.\nsasl.mechanism\u003dSCRAM-SHA-512\nsecurity.protocol\u003dSASL_PLAINTEXT\n\nfor java apps, the following\n\nsasl.jaas.config\u003dorg.apache.kafka.common.security.scram.ScramLoginModule required username\u003d\"user\" password\u003d\"pass\";\n\nFor python, not sure.\nSomething along these lines maybe:\n\n \u0027sasl.username\u0027: SASL_USERNAME,\n \u0027sasl.password\u0027: SASL_PASSWORD,",
      "parentUuid": "e5aa0eea_ceec9e75",
      "range": {
        "startLine": 565,
        "startChar": 12,
        "endLine": 565,
        "endChar": 16
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "decdb77c_3f88fbab",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 565,
      "author": {
        "id": 8012
      },
      "writtenOn": "2023-03-29T17:24:25Z",
      "side": 1,
      "message": "There exist certain issue from python library, \nResults in SCRAM-SHA-512 mechanism handshake failed: Broker: Request not valid in current SASL state: broker\u0027s supported mechanisms\n\nrefered https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/sasl_producer.py\n\nfollowing code used\n-------------------\nSECURITY_PROTOCOL \u003d \"SASL_PLAINTEXT\"\nSASL_MECHANISM \u003d \"SCRAM-SHA-512\"\nSERVERS \u003d \"kafka:9092\"\nSASL_USERNAME \u003d \"admin\"\nSASL_PASSWORD \u003d \"1243\"\n\nconf \u003d {\n\n \u0027bootstrap.servers\u0027: SERVERS,\n \u0027sasl.mechanism\u0027: SASL_MECHANISM,\n \u0027security.protocol\u0027: SECURITY_PROTOCOL,\n \u0027sasl.username\u0027: SASL_USERNAME,\n \u0027sasl.password\u0027: SASL_PASSWORD,\n \u0027group.id\u0027: \"222\",\n \u0027auto.offset.reset\u0027: \u0027smallest\u0027 \n }\nproducer \u003d Producer(conf)\ntestData \u003d \"test\"\nproducer.produce(\u0027PERFORMANCE_MEASUREMENTS\u0027, value\u003dtestData.encode(\u0027utf-8\u0027))\nproducer.poll(1)\n\n\nFollowing are the error logs\n-------------------------------\nkafka               | [2023-03-29 16:10:19,381] INFO [broker-1-to-controller-send-thread]: Recorded new controller, from now on will use broker kafka:9092 (id: 1 rack: null) (kafka.server.BrokerToControllerRequestThread)\nkafka               | [2023-03-29 16:10:24,346] INFO [Controller id\u003d1] Processing automatic preferred replica leader election (kafka.controller.KafkaController)\nkafka               | [2023-03-29 16:10:24,347] TRACE [Controller id\u003d1] Checking need to trigger auto leader balancing (kafka.controller.KafkaController)\nml-prediction-ms    | %4|1680106276.260|CONFWARN|rdkafka#producer-1| [thrd:app]: Configuration property group.id is a consumer property and will be ignored by this producer instance\nml-prediction-ms    | %4|1680106276.260|CONFWARN|rdkafka#producer-1| [thrd:app]: Configuration property auto.offset.reset is a consumer property and will be ignored by this producer instance\nml-prediction-ms    | %3|1680106276.279|FAIL|rdkafka#producer-1| [thrd:sasl_plaintext://kafka:9092/bootstrap]: sasl_plaintext://kafka:9092/bootstrap: SASL SCRAM-SHA-512 mechanism handshake failed: Broker: Request not valid in current SASL state: broker\u0027s supported mechanisms:  (after 0ms in state AUTH_HANDSHAKE)\nml-prediction-ms    | %3|1680106276.286|FAIL|rdkafka#consumer-2| [thrd:sasl_plaintext://kafka:9092/bootstrap]: sasl_plaintext://kafka:9092/bootstrap: SASL SCRAM-SHA-512 mechanism handshake failed: Broker: Request not valid in current SASL state: broker\u0027s supported mechanisms:  (after 1ms in state AUTH_HANDSHAKE)",
      "parentUuid": "92d323a0_44cf7e33",
      "range": {
        "startLine": 565,
        "startChar": 12,
        "endLine": 565,
        "endChar": 16
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    },
    {
      "key": {
        "uuid": "519cc5f3_64edb178",
        "filename": "components/ml-prediction-ms/src/run.py",
        "patchSetId": 38
      },
      "lineNbr": 565,
      "author": {
        "id": 4405
      },
      "writtenOn": "2023-03-29T18:14:41Z",
      "side": 1,
      "message": "Looks like the kafka broker in docker compose isn\u0027t configured with the correct supported mechanisms",
      "parentUuid": "decdb77c_3f88fbab",
      "range": {
        "startLine": 565,
        "startChar": 12,
        "endLine": 565,
        "endChar": 16
      },
      "revId": "9f7c207d16fee8e7f3070d4abcfbdcd27309ce8f",
      "serverId": "14b4e41f-c4e3-4fb9-9955-6a9b5656895a",
      "unresolved": false
    }
  ]
}